Add Generic instance to Entity type

Hi.

I want to use persistent with servant and servant-purescript - package that automatically generates datatypes and utils for fetching them for the client. 

I want to send Entity instances from backend, since I need to have access to ids of sent entities, but it is required by the servant-purescript library that every datatype which will be sent from server to client is an instance of Generic typeclass.

Is it possible to add such typeclass instance to Entity? It would automate much of the work - 
currently I need to define my own Generic types and then map Entity instances returned by sql queries before sending them to client.

Thanks in advance.

@parsonsmatt Is that actually possible? If I add code like:

```haskell
deriving instance (Generic rec) => Generic (Entity rec)
```

or 

```haskell
deriving instance (PersistEntity record, Generic (Key record), Generic record) => Generic (Entity record)
```

to `PersistEntity.hs`, I get:

```
/Users/maximiliantagher/Documents/Clones/Haskell/yesodweb/persistent/persistent/Database/Persist/Class/PersistEntity.hs:183:1: error:
    • Can't make a derived instance of ‘Generic (Entity rec)’:
        Entity must be a vanilla data constructor
    • In the stand-alone deriving instance for
        ‘(Generic rec) => Generic (Entity rec)’
```

Based on googling, you can't derive Generic for Existential Types: https://ghc.haskell.org/trac/ghc/ticket/10514

--- a/persistent/Database/Persist/Class/PersistEntity.hs
+++ b/persistent/Database/Persist/Class/PersistEntity.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE CPP #-}
+{-# LANGUAGE DeriveGeneric #-}
 {-# LANGUAGE ExistentialQuantification #-}
 {-# LANGUAGE TypeFamilies #-}
 {-# LANGUAGE OverloadedStrings #-}
@@ -43,6 +44,7 @@ import Data.Monoid (mappend)
 import qualified Data.HashMap.Strict as HM
 import Data.Typeable (Typeable)
 import Data.Maybe (isJust)
+import GHC.Generics
 
 -- | Persistent serialized Haskell records to the database.
 -- A Database 'Entity' (A row in SQL, a document in MongoDB, etc)
@@ -170,10 +172,11 @@ data Filter record = forall typ. PersistField typ => Filter
 -- your query returns two entities (i.e. @(Entity backend a,
 -- Entity backend b)@), then you must you use @SELECT ??, ??
 -- WHERE ...@, and so on.
-data Entity record = PersistEntity record =>
+data Entity record =
     Entity { entityKey :: Key record
            , entityVal :: record }
 
+deriving instance (PersistEntity record, Generic (Key record), Generic record) => Generic (Entity record)
 deriving instance (PersistEntity record, Eq (Key record), Eq record) => Eq (Entity record)
 deriving instance (PersistEntity record, Ord (Key record), Ord record) => Ord (Entity record)
 deriving instance (PersistEntity record, Show (Key record), Show record) => Show (Entity record)

